pyuvdata/uvdata/ms.py

```diff
@@ -126,11 +886,715 @@ def find_not_ascii(s):
+
+        for key in sorted(data_dict.keys()):
+            blt_idx = [
+                blt_dict[(time, ant1, ant2)]
+                for time, ant1, ant2 in zip(
+                    data_dict[key]["TIME"],
+                    data_dict[key]["ANTENNA1"],
+                    data_dict[key]["ANTENNA2"],
+                )
+            ]
+
+            data_dict[key]["BLT_IDX"] = np.array(blt_idx, dtype=int)
+            data_dict[key]["NBLTS"] = len(blt_idx)
+
+            pol_idx = np.intersect1d(
+                pol_list,
+                data_desc_dict[key]["CORR_TYPE"],
+                assume_unique=True,
+                return_indices=True,
+            )[1]
+            data_dict[key]["POL_IDX"] = pol_idx.astype(int)
+
+        # We have all of the meta-information linked the various data desc IDs,
+        # so now we can finally get to the business of filling in the actual data.
+        data_array = np.zeros((nblts, nfreqs, npols), dtype=complex)
+        nsample_array = np.ones((nblts, nfreqs, npols))
+        flag_array = np.ones((nblts, nfreqs, npols), dtype=bool)
+
+        # We will also fill in our own metadata on a per-blt basis here
+        time_arr = np.zeros(nblts)
+        int_arr = np.zeros(nblts)
+        ant_1_arr = np.zeros(nblts, dtype=int)
+        ant_2_arr = np.zeros(nblts, dtype=int)
+        field_arr = np.zeros(nblts, dtype=int)
+        uvw_arr = np.zeros((nblts, 3))
+
+        # Since each data description (i.e., spectral window) record can technically
+        # have its own values for time, int-time, etc, we want to check and verify that
+        # the values are consistent on a per-blt basis (since that's the most granular
+        # pyuvdata can store that information).
+        has_data = np.zeros(nblts, dtype=bool)
+
+        arr_tuple = (time_arr, int_arr, ant_1_arr, ant_2_arr, field_arr, uvw_arr)
+        name_tuple = ("TIME", "INTERVAL", "ANTENNA1", "ANTENNA2", "FIELD_ID", "UVW")
+        for key in data_dict.keys():
+            # Get the indexing information for the data array
+            blt_idx = data_dict[key]["BLT_IDX"]
+            startchan = data_dict[key]["STARTCHAN"]
+            stopchan = data_dict[key]["STOPCHAN"]
+            pol_idx = data_dict[key]["POL_IDX"]
+
+            # Identify which values have already been populated with data, so we know
+            # which values to check.
+            check_mask = has_data[blt_idx]
+            check_idx = blt_idx[check_mask]
+
+            # Loop through the metadata fields we intend to populate
+            for arr, name in zip(arr_tuple, name_tuple):
+                if np.any(check_mask):
+                    if not np.all(data_dict[key][name][check_mask] == arr[check_idx]):
+                        # This is presently a warning, although it could conceivably
+                        # be made into an error. If doing so, we probably want to change
+                        # the above to 'is_close' rather than strict equality.
```

Comment:
Hmm. One part of me prefers warnings on read because they don't totally block users, but I worry that warnings are easily ignored. What value gets filled in if they vary? The first value? The most common value?

Another option would be to have a keyword that allows the user to turn this into a warning rather than an error but it defaults to erroring so that the user has to make a choice to ignore it.

Reply:
I think I might prefer the latter option (making this an error rather than a warning that can be switched off). There are probably some clever things that we can do about chosing the "best" value, but I might just prefer to default to the first one for now (since it's fastest), and worry about something smarter further down the road.

Reply:
Updated in 3068e067.